From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_git@polytechnique.org>
Date: Sat, 28 May 2016 09:23:41 +0200
Subject: [PATCH] {BUG in 4.7-rc1} ceph: use != instead of |=

gcc complains:

    fs/ceph/dir.c:600:22: error: using the
      result of an assignment as a condition without parentheses
      [-Werror,-Wparentheses]
        } else if (fi->frag |= fpos_frag(new_pos)) {
                   ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
    fs/ceph/dir.c:600:22: note: place
      parentheses around the assignment to silence this warning
        } else if (fi->frag |= fpos_frag(new_pos)) {
                            ^
                   (                             )
    fs/ceph/dir.c:600:22: note: use '!=' to
      turn this compound assignment into an inequality comparison
        } else if (fi->frag |= fpos_frag(new_pos)) {
                            ^~
                            !=
---
 fs/ceph/dir.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/ceph/dir.c b/fs/ceph/dir.c
index 6e0fedf6713b..cf22f1e38fa3 100644
--- a/fs/ceph/dir.c
+++ b/fs/ceph/dir.c
@@ -597,7 +597,7 @@ static bool need_reset_readdir(struct ceph_file_info *fi, loff_t new_pos)
 	if (is_hash_order(new_pos)) {
 		/* no need to reset last_name for a forward seek when
 		 * dentries are sotred in hash order */
-	} else if (fi->frag |= fpos_frag(new_pos)) {
+	} else if (fi->frag != fpos_frag(new_pos)) {
 		return true;
 	}
 	rinfo = fi->last_readdir ? &fi->last_readdir->r_reply_info : NULL;
-- 
